silver/models/transactions.py

```diff
@@ -29,3 +39,32 @@ def customer(self):
     @property
     def payment_processor(self):
         return self.payment_method.payment_processor
+
+    @transition(field=status, source='*', target=State.Canceled)
+    def cancel(self):
+        """
+        The transaction is canceled.
+        """
+        self.payment.fail()
+
+    @transition(field=status, source=State.Uninitialized, target=State.Pending,
+                conditions=[lambda t: t.payment.status == t.payment.Status.Unpaid])
+    def pending(self):
```

Comment:
Rename this to `process` or some other action. `pending` is the state, but not the transition action.